test/edn_query_language/core_test.cljc

```diff
@@ -148,6 +148,56 @@
                          :query        [:sub-query],
                          :children     [{:type :prop, :dispatch-key :sub-query, :key :sub-query}]}]}))))
 
+(deftest test-query->ast-meta
+  (is (= (eql/query->ast ^:root-meta [])
+         {:type :root, :children [], :meta {:root-meta true}}))
+
+  (is (= (eql/query->ast [^:join-meta {:join [:with-meta]}])
+         {:type :root,
+          :children [{:type :join,
+                      :dispatch-key :join,
+                      :key :join,
+                      :query [:with-meta],
+                      :meta {:join-meta true},
+                      :children [{:type :prop, :dispatch-key :with-meta, :key :with-meta}]}]}))
+
+  (is (= (eql/query->ast [^:component {:join [:with-meta]}])
```

Comment:
I think you still need component at the top. Union queries are maps.

Reply:
Unions is under consideration if we decide to go though, but I'm starting to consider if we need the change (check comments in the PR).

About the component, if you try on EQL today, putting the meta at the join (instead of the query) will not add `:component` to the AST. So I think we can keep as-is instead of leaking more of the `:component` out. Wdyt?